File: bot.js
require('dotenv').config();
const axios = require('axios');
const TelegramBot = require('node-telegram-bot-api');

// Загружаем API-ключи из переменных окружения
const TELEGRAM_BOT_TOKEN = process.env.TELEGRAM_BOT_TOKEN;
const TELEGRAM_CHAT_ID = process.env.TELEGRAM_CHAT_ID;

const COINGECKO_API_URL = "https://api.coingecko.com/api/v3/simple/price";
const COINS = ["ethereum", "solana", "act-i-the-ai-prophecy", "pepe", "bonk"]; // Укажи нужные монеты
const CURRENCY = "usd";

const bot = new TelegramBot(TELEGRAM_BOT_TOKEN);

async function getCryptoPrices() {
    let prices = {};
    try {
        const response = await axios.get(COINGECKO_API_URL, {
            params: {
                ids: COINS.join(","), // Запрос сразу для нескольких монет
                vs_currencies: CURRENCY
            }
        });

        for (const coin of COINS) {
            if (response.data[coin]) {
                prices[coin] = parseFloat(response.data[coin][CURRENCY]);
            } else {
                console.warn(`⚠️ Монета ${coin} не найдена на CoinGecko`);
            }
        }
    } catch (error) {
        console.error('❌ Ошибка при получении данных с Binance:', error);
    }
    return prices;
}

async function sendCryptoUpdate() {
    console.log("🚀 Бот запущен!");

    const prices = await getCryptoPrices();
    if (Object.keys(prices).length === 0) {
        //await bot.sendMessage(TELEGRAM_CHAT_ID, '⚠️ Ошибка получения данных с Binance.');
        return;
    }

    let message = '📊 *Криптосводка:*\n\n' +
        Object.entries(prices).map(([symbol, price]) => `${symbol}: ${price} USDT`).join('\n');

    await bot.sendMessage(TELEGRAM_CHAT_ID, message, { parse_mode: 'Markdown' });
}

sendCryptoUpdate();

File: bot.py
import os
import requests
import telebot

# Получаем токены из GitHub Secrets
TELEGRAM_BOT_TOKEN = os.getenv("TELEGRAM_BOT_TOKEN")
TELEGRAM_CHAT_ID = os.getenv("TELEGRAM_CHAT_ID")

# API Binance (публичный)
BINANCE_API_URL = "https://api.binance.com/api/v3/ticker/price"

# Список монет, которые отслеживаем
COINS = ["ACTUSDT", "ETHUSDT", "SOLUSDT"]  # Добавь свои монеты

# Инициализация бота
bot = telebot.TeleBot(TELEGRAM_BOT_TOKEN)

def get_crypto_prices():
    prices = {}
    for coin in COINS:
        try:
            response = requests.get(f"{BINANCE_API_URL}?symbol={coin}")
            response.raise_for_status()
            data = response.json()
            prices[coin] = float(data["price"])
        except requests.RequestException as e:
            print(f"Ошибка при получении {coin}: {e}")
    return prices

def send_crypto_update():
    prices = get_crypto_prices()
    if not prices:
        bot.send_message(TELEGRAM_CHAT_ID, "⚠️ Ошибка получения данных с Binance.")
        return

    message = "📊 *Криптосводка на сегодня:*\n\n"
    message += "\n".join([f"💰 {coin}: `{price}` USDT" for coin, price in prices.items()])

    bot.send_message(TELEGRAM_CHAT_ID, message, parse_mode="Markdown")

if __name__ == "__main__":
    send_crypto_update()  # Одиночный запуск
